app/models/job.rb

```diff
@@ -1,2 +1,4 @@
 class Job < ApplicationRecord
+ # validates :status, inclusion: { in: %w(success failed),
```

Comment:
Unless there's a reason lets omit these commented out lines.

Reply:
It seems that we should put the logic in the model. But code in the job.rb will return all the errors with the customize the message and the error code won't be 400. If it's ok to leave everything in the controller, I will remove the change in job.rb.

Reply:
That's a good point.  Try putting the logic in the model, then in the controller testing if it's valid then passing the validation message with the 400 response.  

There's some documentation here: http://guides.rubyonrails.org/active_record_validations.html

We definitely want to return a 400 if the resource is invalid, and a 500 if some other error (DB access or something) prevented "@job.save" from returning true.